src/eureka/S1_detector_processing/group_level.py

```diff
@@ -125,19 +134,24 @@ def mask_trace(input_model, log, meta):
         smooth_coms[:meta.ignore_low] = smooth_coms[meta.ignore_low]
     if meta.ignore_hi is not None:
         smooth_coms[meta.ignore_hi:] = smooth_coms[meta.ignore_hi]
-    
-    # now create mask based on smooth_coms center.                                                                                                                                                                                                         
-    for nc in range(ncol):
-        mask_low = np.nanmax([0, smooth_coms[nc]-meta.expand_mask])
-        mask_hih = np.nanmin([smooth_coms[nc]+meta.expand_mask, nrow-1])
 
-        trace_mask[mask_low:mask_hih, nc] = np.nan
+    # now create mask based on smooth_coms center.
+    if meta.bg_x1 is not None and meta.bg_x2 is not None:
+        for nc in range(meta.bg_x1, meta.bg_x2):
+            mask_low = np.nanmax([0, smooth_coms[nc]-meta.expand_mask])
+            mask_hih = np.nanmin([smooth_coms[nc]+meta.expand_mask, nrow-1])
+            trace_mask[mask_low:mask_hih, nc] = np.nan
+    else:
+        for nc in range(ncol):
+            mask_low = np.nanmax([0, smooth_coms[nc]-meta.expand_mask])
+            mask_hih = np.nanmin([smooth_coms[nc]+meta.expand_mask, nrow-1])
+            trace_mask[mask_low:mask_hih, nc] = np.nan
```

Comment:
Rather than duplicating code, let's make a new variable `cols` or something, set it equal to `range(meta.bg_x1, meta.bg_x2)` in the `if` statement, and set it to `range(ncol)` in the `else` statement. Then do `for nc in cols:` outside of the if/else statements